src/Microsoft.DotNet.Darc/DarcLib/Helpers/ManifestHelper.cs

```diff
@@ -102,4 +113,73 @@ List<string> GetTargetPaths(DownloadedAsset asset)
         }
         return JObject.FromObject(manifestObject, JsonSerializer.CreateDefault(new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }));
     }
+
+    private static List<DownloadedAsset> SelectMergedManifestAssets(IEnumerable<DownloadedBuild> downloadedBuilds)
+    {
+        return downloadedBuilds.Select(build =>
+            build.DownloadedAssets.Where(asset =>
+                asset.Asset.Name.EndsWith(MergedManifestFileName)))
+            .SelectMany(x => x).ToList();
+    }
+
+    private static Dictionary<string, string> RetrieveAssetOwnerMap(IEnumerable<DownloadedAsset> mergedManifests, ILogger logger)
+    {
+        Dictionary<string, string> assetOwnerMap = new();
+
+        foreach (var mergedManifest in mergedManifests)
+        {
+            XDocument document = null;
+            try
+            {
+                document = XDocument.Load(mergedManifest.UnifiedLayoutTargetLocation);
+            }
+            catch (Exception ex)
+            {
+                logger.LogWarning($"Warning: Loading '{mergedManifest.UnifiedLayoutTargetLocation}' failed with exception: {ex}.");
+                continue;
+            }
+
+            XElement buildElement = document.Element("Build");
+
+            string buildName = buildElement?.Attribute("Name")?.Value;
+            if (buildName == null)
+                continue;
+
+            string repoName = buildName.Replace("dotnet-", string.Empty);
+
+            foreach (var asset in buildElement?.Elements())
+            {
+                string assetName = asset.Attribute("Id")?.Value;
+                if (assetName == null)
+                    continue;
+
+                string assetOwner = asset.Attribute("Owner")?.Value;
+                if (assetOwner == null)
```

Comment:
FYI there's also this:  
`assetOwner ??= repoName;`

But not sure you want to use it here, since there's the comment in the `if` so it's also explicit like that.

Reply:
This suggestion is nice looking and I can drop a comment on top of the line.